### PR TITLE
Fix #5, Changed behaviour of delete key

### LINKED ISSUE #5: readline() alternative that takes a timeout
Suggested by @tpodowd

### PR DESCRIPTION
**Pull request by wader**
* Changed current delete to CharEOT, added new CharDelete key

wader notes:
Don't expect more char after ex esc rune
Kick reader even if no char is deleted

### CODE REVIEW

**Comment by wader** [comment]
Needs cleanup and can't find the repo where i found this fix. Also i've fixed a bug in the original commit where deleting at the end of the line would cause readline to stop writing outpu at all.

**Comment by wader** [comment]
Also related https://github.com/wader/readline/commit/2342177712b561aa1a290fc16c52b6f3a028f614 (Support for windows delete key)

**Comment by wader** [comment]
Rebased on master. Haven't been able to test things as i'm on laptop with no del key, will test later.

Also have to remember what it was that this actually fixed, i think it made the del key work properly in some cases, but it might depend on how the terminal is configured which is confusing. I also had some changes to fix a bug that it introduced (when deleting last on a line when there is no character to delete)  related to kick reader. But after rebase on master i see that that stuff has been reworked so maybe it got fixed?

**Comment by slingamn** [comment]
Yep, got rid of public `KickRead` in #16 :-)

Sorry, what does EOT stand for? Also, what is the intended change in behavior here?

**Comment by wader** [comment]
I honestly don't remember what it actually fixed now but i do remember that it had to do with del-key/ctrl-d behavior and that del-key on an empty readline should not exit the prompt. What makes it confusing is that what terminals sends for del-key/ctrl-d seems to diff and is configurable.

EOT is probably (ASCII?) end-of-transmission in this context.

Will try to find the original commit, maybe that will shine some light also. I mess up so the original author got lost also.

**Comment by wader** [comment]
The original author is @stdiopt but can't find repo where is originated from :(

**Comment by slingamn** [comment]
>del-key on an empty readline should not exit the prompt.

Thanks, I can reproduce this --- I'll treat this as a bug report and then see how to adapt this patch to achieve the desired outcome.

**Comment by wader** [comment]
Great!

**Comment by slingamn** [comment]
I finally understand this patch! I made a minor change, I'll probably merge this soon.

File: terminal.go [hunk 1]
```diff
@@ -358,7 +358,7 @@ func (t *terminal) consumeANSIEscape(buf *bufio.Reader) (result readResult, err
 		r = CharLineEnd
 	case '~':
 		if initial == '[' && data == "3" {
-			r = CharDelete // ???
+			r = MetaDeleteKey // this is the key typically labeled "Delete"
```
Comment (wader) [inline terminal.go:361]: Move comment to const?

**Comment by wader** [comment]
Hehe, don't think i remember how it worked anymore :) it usually take some time to get into terminal mode to remember these things.

**Comment by slingamn** [comment]
This is a pretty good summary FWIW https://gist.github.com/CMCDragonkai/e45d062aad3be6a3be26da76694fdfbf

File: terminal.go (hunk 1 above)
Comment (slingamn) [reply terminal.go:361]: This seems OK to me...the escape sequence parsing code in terminal.go is where I typically get confused

**Comment by wader** [comment]
LGTM

**Pull request merged**